core/domain/services/admin/registrations/list_table/page_header/EventFilterHeader.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace EventEspresso\core\domain\services\admin\registrations\list_table\page_header;
+
+use EE_Admin_Page;
+use EE_Error;
+use EE_Event;
+use EEM_Event;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use EventEspresso\core\services\admin\AdminPageHeaderDecorator;
+use EventEspresso\core\services\request\RequestInterface;
+use InvalidArgumentException;
+use ReflectionException;
+
+/**
+ * Class EventFilterHeader
+ * uses Decorator pattern to add name and link for currently filtered event to admin page header text
+ *
+ * @package EventEspresso\core\domain\services\admin\registrations\list_table\page_header
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class EventFilterHeader extends AdminPageHeaderDecorator
+{
+
+    /**
+     * @var EEM_Event $event_model
+     */
+    private $event_model;
+
+
+    /**
+     * EventFilterHeader constructor.
+     *
+     * @param RequestInterface $request
+     * @param EEM_Event        $event_model
+     */
+    public function __construct(RequestInterface $request, EEM_Event $event_model)
+    {
+        parent::__construct($request);
+        $this->event_model = $event_model;
+    }
+
+
+    /**
+     * @param string $text
+     * @return string
+     * @throws EE_Error
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws InvalidArgumentException
+     * @throws ReflectionException
+     * @since $VID:$
+     */
+    public function getHeaderText($text = '')
+    {
+        $EVT_ID = $this->request->getRequestParam('EVT_ID');
+        $EVT_ID = $this->request->getRequestParam('event_id', $EVT_ID);
+        $EVT_ID = absint($EVT_ID);
+        if ($EVT_ID) {
+            $event = $this->event_model->get_one_by_ID($EVT_ID);
+            if ($event instanceof EE_Event) {
+                $text .= sprintf(
+                    esc_html__(
+                        '%1$s Viewing registrations for the event: %2$s%3$s',
```

Comment:
no we have never decided which option to use... 
I personally favour using `esc_html_x()` because it is more formalized (more well-structured as you stated) but I'm thinking I read somewhere that using `//translators` comments was better for some reason... but I could be thinking of what you said about phpcs...

doesn't matter to me, so if you want to choose...

Reply:
esc_html_x() is OK if it’s a new string
if it’s an existing string, changing to esc_html_x() will break all existing translations

Reply:
thanks for the reminder. No `esc_html_x()` isn't an option when changing strings